File: Day12/Medium/Solution.java
class Solution {
    public int maxProfit(int[] prices) {
       int max = 0;
        int min = Integer.MAX_VALUE;
        for (int i = 0; i < prices.length; i++) {
            min = Math.min(min, prices[i]);
            int pr = prices[i] - min;
            max = Math.max(pr, max);
        }
        return max;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] prices = {7, 1, 5, 3, 6, 4};
        System.out.println(sol.maxProfit(prices)); // Expected: 5
    }
}

File: Day12/Medium/Solution.py
class Solution:
    def maxProfit(self, prices):
        max = 0
        min_val = float('inf')
        for price in prices:
            min_val = min(min_val, price)
            pr = price - min_val
            max = max if pr < max else pr
        return max

if __name__ == "__main__":
    sol = Solution()
    prices = [7, 1, 5, 3, 6, 4]
    print(sol.maxProfit(prices))  # Expected: 5
